src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateProjectDependenciesRule.cs

```diff
@@ -0,0 +1,95 @@
+    {
+        private readonly ITransformApplicator _transformApplicator;
+        private string _projectDirectory;
+
+        public MigrateProjectDependenciesRule(TransformApplicator transformApplicator = null)
+        {
+            _transformApplicator = transformApplicator ?? new TransformApplicator();
+        }
+
+        public void Apply(MigrationSettings migrationSettings, MigrationRuleInputs migrationRuleInputs)
+        {
+            _projectDirectory = migrationSettings.ProjectDirectory;
+
+            var csproj = migrationRuleInputs.OutputMSBuildProject;
+            var projectContext = migrationRuleInputs.DefaultProjectContext;
+            var projectExports = projectContext.CreateExporter("_").GetDependencies(LibraryType.Project);
+
+            var projectDependencyTransformResults =
+                projectExports.Select(projectExport => ProjectDependencyTransform.Transform(projectExport));
+            var propertyTransformResults = new []
+            {
+                AutoUnifyTransform.Transform(true),
+                DesignTimeAutoUnifyTransform.Transform(true)
+            };
+
+            if (projectDependencyTransformResults.Any())
+            {
+                // Use a new item group for the project references, but the common for properties
+                var propertyGroup = migrationRuleInputs.CommonPropertyGroup;
+                var itemGroup = csproj.AddItemGroup();
+
+                foreach (var projectDependencyTransformResult in projectDependencyTransformResults)
+                {
+                    _transformApplicator.Execute(projectDependencyTransformResult, itemGroup);
+                }
+
+                foreach (var propertyTransformResult in propertyTransformResults)
+                {
+                    _transformApplicator.Execute(propertyTransformResult, propertyGroup);
+                }
+            }
+
+        }
+
+        private AddPropertyTransform<bool> AutoUnifyTransform => new AddPropertyTransform<bool>(
+            "AutoUnify",
+            "true",
+            b => true);
+
+        private AddPropertyTransform<bool> DesignTimeAutoUnifyTransform => new AddPropertyTransform<bool>(
+            "DesignTimeAutoUnify",
+            "true",
+            b => true);
+
+        private AddItemTransform<LibraryExport> ProjectDependencyTransform => new AddItemTransform<LibraryExport>(
+            "ProjectReference",
+            export => 
+            {
+                if (!export.Library.Resolved)
+                {
+                    throw new Exception("Cannot migrate unresolved project dependency, please ensure restore has been run.");
+                }
+
+                var projectFile = ((ProjectDescription)export.Library).Project.ProjectFilePath;
+                var projectDir = Path.GetDirectoryName(projectFile);
+                var migratedProjectFileName = Path.GetFileName(projectDir) + ".csproj";
```

Comment:
I should've mentioned this in the PR notes, I want the ability to recursively migrate projects in a future PR.

I haven't done any thinking about migrating `F#` projects yet, I imagine they'll be a lot of the same with some differences. I can do a quick check for the compilerName on the second project here though to get the correct extension. 

I'll write this down for the next PR though.